### PR TITLE
5 get articles

### PR DESCRIPTION
**Pull request by hc-tbhc**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -102,4 +91,47 @@ describe('/api', () => {
...
+    test('200: GET /api/articles - responds with an array of article objects containing the relevant keys', () => {
+        return request(app)
+        .get('/api/articles')
+        .expect(200)
+        .then(({ body }) => {
+            const {articles} = body;
+            expect(articles.length > 0).toBe(true);
 
+            articles.forEach((article) =>{
+                expect(article).toMatchObject({
```
Comment (roisinoreilly) [inline __tests__/app.test.js:104]: excellent

File: __tests__/app.test.js [hunk 2]
```diff
@@ -102,4 +91,47 @@ describe('/api', () => {
...
 })
 
+describe('/api/articles', () => {
+    test('200: GET /api/articles - responds with an array of article objects containing the relevant keys', () => {
+        return request(app)
+        .get('/api/articles')
+        .expect(200)
+        .then(({ body }) => {
+            const {articles} = body;
+            expect(articles.length > 0).toBe(true);
```
Comment (roisinoreilly) [inline __tests__/app.test.js:101]: this is fine but there's a few other jest matchers you can use here like .toHaveLength, .toBeGreaterThan, but I would personally just assert against the actual length itself here

File: __tests__/app.test.js [hunk 3]
```diff
@@ -102,4 +91,47 @@ describe('/api', () => {
...
+        return request(app)
+        .get('/api/articles')
+        .expect(200)
+        .then(({ body }) => {
+            const {articles} = body;
+            const dates = [];
+            expect(articles.length > 0).toBe(true);
+
+            articles.forEach((article) =>{
+                dates.push(article.created_at);
```
Comment (roisinoreilly) [inline __tests__/app.test.js:129]: not quite sure what this is for but you definitely don't need it

File: __tests__/app.test.js [hunk 4]
```diff
@@ -102,4 +91,47 @@ describe('/api', () => {
...
+            })
+
+            expect(articles).toBeSortedBy('created_at', {
```
Comment (roisinoreilly) [inline __tests__/app.test.js:132]: fab

File: controllers/controller.js [hunk 5]
```diff
@@ -21,4 +21,17 @@ function retrieveArticleById(req, res, next) {
     })
 }
 
-module.exports = { retreiveTopics, retrieveArticleById };
\ No newline at end of file
+function retrieveArticles(req, res, next) {
+    getArticlesSorted()
+    .then((result) => {
+        const articlesToSend = result.map((article) => {
+            delete article.body;
```
Comment (roisinoreilly) [inline controllers/controller.js:28]: you shouldn't be deleting anything here, you should just use the model to only select what you want

**Review by roisinoreilly — COMMENTED** [review]
nice work so far on this ticket and good testing and error handling, there's just a few odd things I'd like you to come back to and have a look at at some point (none of them urgent, just whenever you get the chance to revisit tickets). Have a look through my comments but any questions about them let me know

File: __tests__/app.test.js (hunk 3 above)
Comment (hc-tbhc) [reply __tests__/app.test.js:129]: I think I was thinking of another way to do the test but being sloppy I left this in before I decided to do it the way I did it, I'll come back to this one and straighten out the problems in my code here

**Pull request merged**